src/hydrate.ts

```diff
@@ -273,6 +274,13 @@ export class Hydratable<T> {
 
   private setObject(key: string, value: any, options: HydrateOptions) {
     if (!options.allowNull && value === null) { return; }
+    const type = options.dictionaryValueType;
+    if (type && type instanceof Function) {
+      Object.keys(value).forEach(key => {
+        value[key] = new type(value[key]);
+      });
+      this.setOnThis(key, this.copyObject(value));
```

Comment:
```suggestion
```

Reply:
Would this change make sense to you? Because this is done down on line 284 anyways.

Reply:
Oh haha good catch! I'll make that change 👍